causal_testing/specification/metamorphic_relation.py

```diff
@@ -214,44 +214,63 @@ def __str__(self):
         )
 
 
-def generate_metamorphic_relations(dag: CausalDAG) -> list[MetamorphicRelation]:
-    """Construct a list of metamorphic relations implied by the Causal DAG.
+def generate_metamorphic_relations(dag: CausalDAG, skip_ancestors: bool) -> list[MetamorphicRelation]:
+    """Construct a list of metamorphic relations based on the DAG or cyclic graph.
```

Comment:
Unsure if actually relevant but in the literature a causal graph with loops is called a DCG (or you could just refer to either as a "causal graph". Pedantic suggestion though